src/watch.rs

```diff
@@ -84,14 +84,14 @@ impl WatchSystem {
             DebouncedEvent::Create(path) | DebouncedEvent::Write(path) | DebouncedEvent::Remove(path) | DebouncedEvent::Rename(_, path) => path,
             _ => return,
         };
-        for path in ev_path.ancestors() {
-            if self.ignores.iter().map(|p| p.as_path()).any(|p| p == path) {
-                return; // Don't emit a notification if ignored.
+
+        // If path is NOT in ignore list
+        if !ev_path.ancestors().any(|path| self.ignores.contains(&path.to_path_buf())) {
+            // Build
+            if let Err(err) = self.build.build().await {
+                self.progress.println(format!("{}", err));
```

Comment:
Two things here. First, the logical changes here look like a solid improvement. Thanks! The main thing that sticks out to me is the `&path.to_path_buf()`. This allocates a new PathBuf. If `.contains` doesn't work with `&path`, then I'm thinking that we should just write our own comparator, something like:

```rust
if !ev_path.ancestors().any(|path| self.ignores.iter().any(|ignore| ignore == path)) {
```

Might have to tweak the use of a `&` here or there in the comparison, but that will save us potentially A BUNCH of allocations on the hot path, as this routine is definitely on the hot path.

Also, though this is less important, just because the negation at the beginning of the `if !ev_path ...` block adds some additional cognitive load, we could just remove that `!` and have the `return; // Don't emit a notification if ignored.` line on the inside as it was, and then move the build invocation to the location it was in previously.

Second, let's either drop the comments, or ensure that they are complete sentences for folks that might be making changes to this in the future `:)`.